runners/flink/src/main/java/org/apache/beam/runners/flink/TestFlinkRunner.java

```diff
@@ -33,6 +33,10 @@
 
   private TestFlinkRunner(FlinkPipelineOptions options) {
     options.setShutdownSourcesOnFinalWatermark(true);
+    if (options.getParallelism() == -1) {
+      // Limit parallelism to 2 to avoid too much memory consumption during local execution
+      options.setParallelism(2);
```

Comment:
I would set it to 1. If tests really have something parallelism related to assert, they can should set it explicitly.

Reply:
Done